### PR TITLE
Add JDK11 image

### PR DESCRIPTION
**Pull request by irishshagua**
* Support both JDK 8 and JDK 11 images
  * Centralise the delivery build logic in a re-usable script
  * Change dynamic mem options script to use JDK 10+ MaxRAMPercentage property

### CODE REVIEW

File: utils/java-dynamic-memory-opts [hunk 1]
```diff
@@ -14,11 +14,16 @@ if [ -z "$MEM_JAVA_PERCENT" ]; then
...
 fi
 
-# MEM_TOTAL_KB can be set from the outside, this is important for container-based infrastructure:
-# /proc/meminfo would return the host/node capacity and not the container's limit
-if [ -z "$MEM_TOTAL_KB" ]; then
-    MEM_TOTAL_KB=$(cat /proc/meminfo | grep MemTotal | awk '{print $2}')
-fi
-MEM_JAVA_KB=$(($MEM_TOTAL_KB * $MEM_JAVA_PERCENT / 100))
+JVERSION=$(java -version 2>&1 | head -n 1 | awk -F '"' '{print $2}')
+if [ $JVERSION == 1.8* ]; then
```
Comment (kingslef) [inline utils/java-dynamic-memory-opts:18]: I don't think you can use pattern matching with plain `[`, but bash's `[[` is needed:
```
$ jversion=1.8.0_181
$ [ $jversion == 1.8* ]; echo $?
1
$ [[ $jversion == 1.8* ]]; echo $?
0
```

but if the file is supposed to be for `sh`, then something else might be needed.

Comment (irishshagua) [reply utils/java-dynamic-memory-opts:18]: I'm not sure tbh...
There didn't seem to be much enthusiasm to accept this PR so I've moved to using non stups base jdk images. If you'd like you can take it over. If not I'll prob close off this PR.

Comment (hjacobs) [reply utils/java-dynamic-memory-opts:18]: Regarding "enthusiasm": I'm just looking for more insights on how people run JDK 11 --- I haven't gotten much feedback so far, so I held back with jumping on the topic.

Comment (otrosien) [reply utils/java-dynamic-memory-opts:18]: Our team is pretty happy with the debian-slim based openjdk image.

Comment (otrosien) [reply utils/java-dynamic-memory-opts:18]: There is also AdoptOpenJDK https://hub.docker.com/r/adoptopenjdk/openjdk11 which provides ubuntu-based images.

**Comment by hjacobs** [comment]
This repository is obsolete. Closing.

**Pull request closed without merging**